NorthstarDedicatedTest/serverauthentication.cpp

```diff
@@ -152,7 +155,9 @@ bool ServerAuthenticationManager::AuthenticatePlayer(void* player, int64_t uid,
 	{
 		// use stored auth data
 		AuthData authData = m_authData[authToken];
-		if (!strcmp(strUid.c_str(), authData.uid)) // connecting client's uid is the same as auth's uid
+
+		if (!strcmp(strUid.c_str(), authData.uid) &&
+			!strcmp(name, authData.name)) // connecting client's uid is the same as auth's uid and name matches supplied one
```

Comment:
This check should be `!strcmp(strUid.c_str(), authData.uid) && (!*authData.name || !strcmp(name, authData.name))` to preserve compatibility with old master servers and to fail open.

Reply:
Fails open now :)